app/post.go

```diff
@@ -17,13 +20,35 @@ import (
 )
 
 var (
-	c = &http.Client{
-		Timeout: 5 * time.Second,
-	}
+	httpClient *http.Client
 
+	httpTimeout       = time.Duration(10 * time.Second)
 	linkWithTextRegex = regexp.MustCompile(`<([^<\|]+)\|([^>]+)>`)
 )
 
+func dialTimeout(network, addr string) (net.Conn, error) {
+	return net.DialTimeout(network, addr, httpTimeout)
+}
+
+func init() {
+	p, ok := os.LookupEnv("http_proxy")
```

Comment:
Linux convention? https://wiki.archlinux.org/index.php/proxy_settings

or rename it to `PROXY_ENV`